src/Counters.hpp

```diff
@@ -22,20 +22,22 @@
 
 #include <vector>
 
+enum class Stats;
+
 // A simple wrapper around a vector of integers
 // used for the statistics counters.
 class Counters
 {
 public:
   Counters();
 
-  unsigned& operator[](size_t index);
-  unsigned operator[](size_t index) const;
+  unsigned& operator[](Stats index);
+  unsigned operator[](Stats index) const;
 
   size_t size() const;
 
   // Return true if all counters are zero, false otherwise.
-  bool all_zero() const;
+  [[nodiscard]] bool all_zero() const;
```

Comment:
Oh I thought I removed it. After adding it I spent some time reading on how people generally handle nodiscard. It seems (as usual) the default is wrong. So now people use it only when they failed at naming. Like empty(). Is it a query or a request? Kind of implied already since the function is const.